src/hardware/classes/Laptop.ts

```diff
@@ -0,0 +1,50 @@
+import CPU from "./CPU";
+import RAM from "./RAM";
+import Storage from "./Storage";
+import Display from "./Display";
+
+export default class Laptop {
+    public readonly id: number;
+    public readonly model: string;
+    public readonly brand: string;
+    public readonly cpu: CPU;
+    public readonly ram: RAM;
+    public readonly storage: Storage[];
+    public readonly screen: Display;
+    // Dimensions of laptop in centimeters
+    public readonly dimensions: {x: number, y: number, z: number};
+    // Weight in kilograms
+    public readonly weight: number;
+    public readonly material: string;
+    public readonly launchDate: Date | null;
+    // Battery capacity in Wh
+    public readonly batteryCapacity: number;
+    public readonly microphonePresent: boolean;
+    // Null means no camera is present
+    public readonly camera: {resolution: number, fps: number} | null;
+    public readonly WiFiConnectivity: {"version": string | number, "dualBand": boolean} | null;
```

Comment:
the wifi bands are included in the standard/specification/version. Some "versions" will only be a different version because e.g. they are dual-band. Otherwise we could make an array with the bands, like 2.4, 5, etc. Next gen WiFi may have up to 3 bands afaik.

```suggestion
    public readonly wifi: {"version": string | number, "dualBand": boolean} | null;
```